File: 品味Java的21種設計模式/dp_java/strategy/example4/OldCustomerStrategy.java
package cn.javass.dp.strategy.example4;
/**
 * 具体算法实现，为老客户计算应报的价格
 */
public class OldCustomerStrategy implements Strategy{
	public double calcPrice(double goodsPrice) {
		System.out.println("对于老客户，统一折扣5%");
		return goodsPrice*(1-0.05);
	}
}

File: 品味Java的21種設計模式/dp_cpp/strategy/example4/OldCustomerStrategy.cpp
#include "OldCustomerStrategy.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace strategy
			{
				namespace example4
				{

					double OldCustomerStrategy::calcPrice(double goodsPrice)
					{
						puts("对于老客户，统一折扣5%");
						return goodsPrice*(1-0.05);
					}
				}
			}
		}
	}
}
